grails-app/domain/org/pih/warehouse/order/Order.groovy

```diff
@@ -518,6 +518,11 @@ class Order implements Serializable {
         return orderType?.isTransferOrder()
     }
 
+    Boolean isFullyInvoiceable() {
+        Set<OrderItem> encumberedOrderItems = orderItems.findAll{ it -> it.encumbered }
+        return encumberedOrderItems && encumberedOrderItems.every{ it -> it.invoiceable }
```

Comment:
Any particular reason to not do it in one loop with something like:
`return !orderItems.any{ it -> it.encumbered && !it.invoiceable }`

Reply:
the example you have provided fails on cases where none of our items is encumbered.
If none of them are encumbered it returns `false` for `.any{}` and a negation returns `true` for `isFullyInvoiceable` which does not make sense which is why I added an additional check for if there are any encumbered items.

Reply:
Last thought: What about adjustments? If we have items that are encumbered and non-invoiceable, but we have an invoiceable adjustment, we'd still want the confirmation to pop up, right?